guardian/riff-raff
# set infinite timeout

## What does this change?

Follows on from #826 which in turn follows on from #740.This first created an optional configuration for the GCS deployment type to delete obsolete files. We found when we merged this that airflow deploys to PROD ( from PROD riffraff ) seemed to error with a `java.net.SocketException`. #826 seemed to almost fix this but the problem re-occurred intemittently but only with scheduled and continual deploys. This change makes the timeout set in #825 infinite and seems to fix the problem. 

I set up temporary CD and SD configurations on  https://riffraff.code.dev-gutools.co.uk/ and immediately recreated the problem. This change fixes it.



## How to test



Deploy main to  https://riffraff.code.dev-gutools.co.uk and then set up a CD configuration using for `ophan:data-lake:airflow` using the branch `clean-code-test` to deploy to CODE on successful build. Kick of a build in [TeamCity](https://teamcity.gutools.co.uk/buildConfiguration/Ophan_DataLake_Airflow?branch=clean-code-test&buildTypeTab=overview&mode=builds) Observe the breaking deploy. It may need a couple tries Repeat to observe consistency. Deploy this branch to RiffRaff Code and rebuild the airflow branch and observe that resulting CDs are successful

## How can we measure success?



Zero failing Airflow Continual and Scheduled deploys in RiffRaff PROD

## Have we considered potential risks?



 I think this deployment type is only used by Data Tech to deploy airflow, and the configuration that caused the problem is [optional configuration](https://github.com/guardian/riff-raff/blob/main/magenta-lib/src/main/scala/magenta/deployment_type/GCS.scala#L64) only used by us on [prod](https://github.com/guardian/ophan-data-lake/blob/main/airflow/riff-raff.yaml#L30)

## Images



## Accessibility
